guides/howtos/test-using-mox.md

```diff
@@ -0,0 +1,88 @@
+# Test Using Mox with Tesla
+
+To mock HTTP requests in your tests using Mox with the Tesla HTTP client,
+follow these steps:
+
+## 1. Define a Mock Adapter
+
+First, define a mock adapter that implements the Tesla.Adapter behaviour. This
+adapter will intercept HTTP requests during testing.
+
+Create a file at `test/support/mocks.ex`:
+
+```elixir
+# test/support/mocks.ex
+Mox.defmock(MyApp.MockAdapter, for: Tesla.Adapter)
+```
+
+## 2. Configure the Mock Adapter for Tests
+
+In your `config/test.exs` file, configure Tesla to use the mock adapter you
+just defined:
+
+```elixir
+# config/test.exs
+config :tesla, adapter: MyApp.MockAdapter
+```
+
+If you are not using the global adapter configuration, ensure that your Tesla
+client modules are configured to use `MyApp.MockAdapter` during tests.
+
+## 3. Set Up Mocking in Your Tests
+
+Create a test module, for example `test/demo_test.exs`, and set up `Mox` to
+define expectations and verify them:
+
+```elixir
+defmodule MyApp.FeatureTest do
+  use ExUnit.Case, async: true
+
+  require Tesla.Test
+
+  setup context, do: Mox.set_mox_from_context(context)
+  setup context, do: Mox.verify_on_exit!(context)
```

Comment:
isn't the "standard" way of doing this something like:


```ex
import Mox

setup :verify_on_exit!
```

? 

I know it's the same, but I think the imported version is what Mox uses in its docs.

Reply:
I am focusing on removing magical things that, although "standard" adds to confusion, especially for newbies;

I expect users to be familiar with `Mox` to some extent, for sure, but also avoid trivial problems I seen before if possible.

Reply:
Give the `import Mox` is the very first thing in [mox docs](https://hexdocs.pm/mox/Mox.html) the `setup context ...` syntax was actually more confusing for me (is something special needed for tesla & mox to work together?).

I'd love to get some more opinions on it.

Reply:
fixing the docs; no strong opinions at all

Reply:
done